src/sentry/api/bases/doc_integrations.py

```diff
@@ -14,12 +14,23 @@
 
 
 class DocIntegrationsPermission(SentryPermission):
+    """ "
+    Allows all org members to access GET as long as they have the necessary
+    scopes. For item endpoints, the doc integration must be published.
+
+    # TODO(schew2381): Remove superuser language once staff feature flag is rolled out
+    Superusers can access unpublished doc integrations (GET) and also use PUT + DEL
+    which endpoints which are all accessible through _admin.
+    """
+
     scope_map = {"GET": PARANOID_GET}
 
     def has_permission(self, request: Request, view: object) -> bool:
         if not super().has_permission(request, view):
             return False
 
+        # TODO(schew2381): Remove superuser check once staff feature flag is rolled out.
+        # We want to allow staff through the StaffPermissionMixin instead of mixing logic here.
```

Comment:
how will you do this through the mixin if the `is_active_superuser` call is inside of `has_permission` and after some other checks? replace `is_active_superuser` with with `is_active_staff`?

Reply:
Superuser shouldn't have access to these endpoints as they'll be used for _admin only

Reply:
ohhh the mixin calls the permission class's `has_object_permission` and `has_permission`. got confused there for a sec